website/blog/2025-03-01-2025-roadmap/index.md

```diff
@@ -0,0 +1,81 @@
+
+For examples:
+
+- OpenDAL hasn't reached version 1.0 yet and still introduces breaking changes from time to time, sometimes even requiring code modifications for an OpenDAL upgrade. This also creates a burden for libraries that depend on OpenDAL, as every breaking change affects them as well.
+- OpenDAL lacks comprehensive documentation, particularly for its bindings in Python and Java. Users need to invest significant effort and love to integrate OpenDAL bindings into their projects.
+- OpenDAL lacks some important features, such as checksum support, caching, metrics of underlying http requests and initialization from a URI.
+
+I believe we should perfect production adoption in 2025 to get OpenDAL ready for early adopters. Only in this way can we implement our vision: **One Layer, All Storage.**
+
+## What to do in 2025?
+
+Our plans for 2025 include the following:
+
+### Features Needed in Production
+
+In 2025, we plan to implement the following features that are importmant for production adoption:
+
+- [Context](https://github.com/apache/opendal/issues/5480): Introduce context in OpenDAL so that services and layers can share the same context. This enables users to gain deeper insights into OpenDAL's internal operations by incorporating metrics, logging, and tracing into the underlying HTTP client we use.
+- [Versioning](https://github.com/apache/opendal/issues/2611): Introduce full file versioning support in OpenDAL, enabling users to read, write, delete, list, and restore versioned files. This functionality will allow users to recover mistakenly deleted files and facilitate disaster recovery.
+- [Checksum](https://github.com/apache/opendal/issues/5549): Introduce end-to-end checksum support in OpenDAL, enabling users to perform checksums during reading and writing without worrying about bit flips in memory or over the network.
+- [Caching](https://github.com/apache/opendal/issues/5678): Provide high-quality built-in cache support in OpenDAL while ensuring users have the flexibility to implement their own caching logic.
+- [Initialization From Uri](https://github.com/apache/opendal/issues/3022): Allow users to initialize OpenDAL from a URI string, making it easier to configure and use OpenDAL in various environments.
+
+### Improvements Needed for Production
+
+In 2025, we plan to improve the following aspects of OpenDAL to ensure users can confidently use it in production.
+
+- Documentation: Improve the documentation for OpenDAL, particularly for bindings like Python, Node.js, and Java. The first step is to generate well-structured documentation for the configuration values of each service.
+- Communitation: Bring Back Our Community Meetings. OpenDAL used to hold regular tri-weekly meetings, but we have been unable to maintain them for years. We now plan to revive these meetings and encourage more face-to-face discussions in the coming years.
```

Comment:
```suggestion
- Communitation: Bring Back Our Community Meetings. OpenDAL used to hold regular tri-weekly meetings, but we have been unable to maintain them in the last year. We now plan to revive these meetings and encourage more face-to-face discussions in the coming years.
```

Reply:
I don't think this is quite easy. If we (I suppose mainly @Xuanwo or any user/developer who hit an issue recently) can prepare an agenda (one week or so) before calling the meeting, we can attract more attendees.